Unable to build since ESPHome 2026.2.0

Hi,

there was some redesign in the build process of esphome 2026.2.0 and since then I am unable to compile for my nspanels.

`src/esphome/components/nspanel_lovelace/nspanel_lovelace.cpp:2: src/esphome/components/nspanel_lovelace/nspanel_lovelace.h:16:10: fatal error: esp_http_client.h: No such file or directory  ******************************************************************************* * Looking for esp_http_client.h dependency? Check our library registry! * * CLI  > platformio lib search "header:esp_http_client.h" * Web  > https://registry.platformio.org/search?q=header:%1B%5Bm%1B%5BKesp_http_client.h * *******************************************************************************     16 | #include <esp_http_client.h>       |          ^~~~~~~~~~~~~~~~~~~ compilation terminated. *** [.pioenvs/nspanel-buegelzimmer/src/esphome/components/nspanel_lovelace/nspanel_lovelace.cpp.o] Error 1 In file included from src/esphome/components/nspanel_lovelace/nspanel_lovelace_upload.cpp:1: src/esphome/components/nspanel_lovelace/nspanel_lovelace.h:16:10: fatal error: esp_http_client.h: No such file or directory  *******************************************************************************`

Probably has something to do with this part of the changelog

"ESP32 Arduino selective compilation: Arduino libraries are now disabled by default. External components that use Arduino libraries must call cg.add_library("LibraryName", None) in their to_code() to enable them."

--- a/components/nspanel_lovelace/__init__.py
+++ b/components/nspanel_lovelace/__init__.py
@@ -2,6 +2,7 @@ from esphome import automation
 import esphome.config_validation as cv
 import esphome.codegen as cg
 from esphome.components import mqtt, uart
+from esphome.components.esp32 import include_builtin_idf_component
 from esphome.const import (
     CONF_ID,
     CONF_TRIGGER_ID,
@@ -84,5 +85,7 @@ async def to_code(config):
     if CORE.is_esp32 and CORE.using_arduino:
         cg.add_library("NetworkClientSecure", None)
         cg.add_library("HTTPClient", None)
+    elif CORE.is_esp32:
+        include_builtin_idf_component("esp_http_client")
 
     cg.add_define("USE_NSPANEL_LOVELACE")

